src/coverage.ts

```diff
@@ -0,0 +1,86 @@
+import { fork, execSync } from 'child_process';
+import path from 'path';
+import fs from 'fs';
+
+export function cleanUp(): void {
+  if (fs.existsSync('./contracts-backup/')) {
+    execSync('cp -rf ./contracts-backup/ ./contracts');
+  }
+  execSync('rm -rf ./contracts-backup/');
+  execSync('rm -rf ./build/contracts/');
+  execSync('rm -rf ./.coverage_artifacts/');
+  execSync('rm -rf ./.coverage_contracts/');
+}
+
+['SIGINT', 'SIGTERM', 'SIGQUIT'].forEach((signal) =>
+  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+  // @ts-ignore
+  process.on(signal, () => {
```

Comment:
I think we also have to register an event handler on the `'exit'` event. See how this is implemented [by the `exit-hook` package](https://github.com/sindresorhus/exit-hook/blob/b504031173d8e2c2ce6bb9f0b6557cbe6dedecd5/index.js#L26-L28).

Reply:
Would you recommend using `exit-hook` package here?

Reply:
Actually yes. I initially didn't recommend it because I was under the impression that users would copy-paste the script.